incubator/hnc/pkg/reconcilers/hnc_config_test.go

```diff
@@ -12,14 +13,24 @@ import (
 	v1 "k8s.io/api/rbac/v1"
 	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
 var _ = Describe("HNCConfiguration", func() {
-	// sleepTime is the time to sleep for objects propagation to take effect.
+	// objectPropagationTime is the time to sleep for objects propagation to take effect.
 	// From experiment it takes ~0.015s for HNC to propagate an object. Setting
 	// the sleep time to 1s should be long enough.
 	// We may need to increase the sleep time in future if HNC takes longer to propagate objects.
-	const sleepTime = 1 * time.Second
+	const objectPropagationTime = 1 * time.Second
+	// hncConfigReconciliationTime is the time to wait for reconciliation of the config
+	// object happens. It is the same as the sleep time at the beginning of the Reconcile method
+	// of HNCConfiguration.
+	const hncConfigReconciliationTime = 3 * time.Second
+	// hncConfigStatusUpdateTime is the time to wait for updating HNCConfiguration status on
+	// the apiserver. From experiment, this is the minimal integer seconds needed for the
+	// status to be updated successfully on the apiserver.
+	// We may need to increase the time in future if it takes longer to update status.
+	const hncConfigStatusUpdateTime = 7 * time.Second
```

Comment:
This seems too high to me. From the time we make our _last_ change to an object, I would expect it would take no more than three seconds plus a little bit extra (e.g. four seconds should be enough): 3s for the HNC Config sleep to end, plus a bit of time for it to do its work. So if we said:

`SetDefaultEventuallyTimeout(hncConfigReconciliationTime + 2 * time.Second)`

Then `Eventually` will wait up to five seconds and that _should_ be more than enough time. (Or don't change the global default and just pass this into each `Eventually`) call.

Do you agree with this logic? And if so, do you know why it doesn't work?

Reply:
Ditto. Resolve this comment since it is outdated.